ui/pages/token-allowance/token-allowance.js

```diff
@@ -94,26 +98,24 @@ export default function TokenAllowance({
   const unapprovedTxCount = useSelector(getUnapprovedTxCount);
   const unapprovedTxs = useSelector(getUnapprovedTransactions);
 
-  const customPermissionAmount = customTokenAmount.toString();
+  let customPermissionAmount = customTokenAmount.toString();
+
+  const maxTokenAmount = calcTokenAmount(MAX_UNSIGNED_256_INT, decimals);
+  if (customTokenAmount.length > 1 && Number(customTokenAmount)) {
+    const customSpendLimitNumber = new BigNumber(customTokenAmount);
+    if (customSpendLimitNumber.greaterThan(maxTokenAmount)) {
+      customPermissionAmount = 0;
+    }
+  }
 
-  const customTxParamsData = customTokenAmount
+  const customTxParamsData = Number(customTokenAmount)
     ? getCustomTxParamsData(data, {
         customPermissionAmount,
         decimals,
       })
     : null;
 
-  let fullTxData = { ...txData };
-
-  if (customTxParamsData) {
```

Comment:
Why is this if block that modifies the `fullTxData` to include `customTxParamsData` being removed?

Reply:
@VSaric do you know why this code is removed?

Reply:
Okay, I can't see VSaric's comment here, but it was explained in slack that this removal was meant to fix a bug whereby if decimal numbers are entered for a custom amount, the approval transaction fails.

The problem with this code removal is that this is the code that adds the custom amount to the transaction data. So if this code is removed, the approval amount that is submitted will be the dapp proposed or default amount, not the custom amount the user entered.

Reply:
the bug this intended to fix is shown in the attached video. The problem is that we are not properly converting decimals numbers with precision < 0 to hex numbers. The proper fix to this bug is to fix that problem.


Uploading Screen Recording 2022-12-20 at 15.59.45.mov…

Reply:
In fact, I think the problem here is that the test case that resulted in the error entered more decimal numbers than the token allows, which causes the problem with a decimal point within the hex value/data.

So the proper fix here is to prevent the user from enter more numbers in the decimal place than the token has decimals.

So for example @VSaric, if the token's decimals are 4 (as is the case of the contract deployment from our test dapp), then the user should not be allow to enter `1.12345` but should be allowed to enter `1.1234`

Can you remove this code change so that `fullTxData` is set as it was before, and then create a separate PR for validating / restricting the number of digits after the decimal point that the user can enter?

Reply:
Yes, I will set `fullTxData` as it was before right away! Thank you @danjm  for clarifying this!

Reply:
@danjm I added `replaceCommaToDot` function and set `fullTxData` as it was before in my latest commit. Now users can enter numbers with comma (`,`) and dot (`.`) and it will work as expected. If users entered, for example, `1,1234` transaction data will change. The approval amount that is submitted will be the custom amount the user entered, so it will adds the custom amount to the transaction data. Done [here](https://github.com/MetaMask/metamask-extension/pull/16676/commits/17e5277ca6c596eaf572ce445866a12adf5d9067).

Reply:
I also started to work for validating / restricting the number of digits after the decimal point that the user can enter like @danjm mention [here](https://github.com/MetaMask/metamask-extension/pull/16676#discussion_r1054314334) and I will create a separate PR for this.